rust rule: Add support of IP route rule

cathay4t opened the pull request:
Integration test cases enabled except
`test_route_rule_add_with_auto_route_table_id` which require support of
DHCP option.

ffmancera reviewed (changes requested):
Please, rebase and check CI.

ffmancera commented on rust/src/lib/ip.rs:
```diff
@@ -449,10 +449,42 @@ pub(crate) fn is_ipv6_addr(addr: &str) -> bool {
+    }
+}
+
+impl std::convert::Into<String> for &InterfaceIpAddr {
+    fn into(self) -> String {
+        format!("{}/{}", &self.ip, self.prefix_length)
+    }
+}
```
I agree with clippy here. Rust documentation recommends always to implement `From<&ip::InterfaceIpAddr>`. IMO, it is more clear for the developers.

cathay4t replied:
Fixed. Thanks.

ffmancera reviewed (changes requested):
It looks good, I made a comment and please check clippy warnings.

cathay4t committed:
rust rule: Add support of IP route rule

Integration test cases enabled except
`test_route_rule_add_with_auto_route_table_id` which require support of
DHCP option.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM!
